### PR TITLE
Fix resource leak

### PR DESCRIPTION
**Pull request by asdf2014**
### Description

Fix resource leak

<hr>

This PR has:
- [x] been self-reviewed.
   - [x] using the [concurrency checklist](https://github.com/apache/incubator-druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

<hr>

### CODE REVIEW

File: core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java [hunk 1]
```diff
@@ -56,14 +56,26 @@
...
+      queue.put(channelStream);
     }
     catch (InterruptedException e) {
       log.error(e, "Queue appending interrupted");
       Thread.currentThread().interrupt();
       throw new RuntimeException(e);
     }
+    finally {
+      if (channelStream != null) {
+        try {
```
Comment (leventov) [inline core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java]: This block is essentially `CloseQuietly.close()`.

File: core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java [hunk 2]
```diff
@@ -116,6 +130,16 @@ public InputStream nextElement()
         Thread.currentThread().interrupt();
         throw new RuntimeException(e);
       }
+      finally {
+        if (channelStream != null) {
+          try {
```
Comment (leventov) [inline core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java]: And here

File: extensions-contrib/ambari-metrics-emitter/src/main/java/org/apache/druid/emitter/ambari/metrics/WhiteListBasedDruidToTimelineEventConverter.java [hunk 3]
```diff
@@ -214,7 +214,9 @@ public TimelineMetric druidEventToTimelineMetric(ServiceMetricEvent serviceMetri
         actualPath = this.getClass().getClassLoader().getResource("defaultWhiteListMap.json").getFile();
         LOGGER.info("using default whiteList map located at [%s]", actualPath);
         InputStream byteContent = this.getClass().getClassLoader().getResourceAsStream("defaultWhiteListMap.json");
```
Comment (leventov) [inline extensions-contrib/ambari-metrics-emitter/src/main/java/org/apache/druid/emitter/ambari/metrics/WhiteListBasedDruidToTimelineEventConverter.java]: Simpler
```java
defaultWhiteListMapUrl = this.getClass().getClassLoader().getResource("defaultWhiteListMap.json");
actualPath = defaultWhiteListMapUrl.getFile();
LOGGER.info(...);
fileContent = Resources.toString(defaultWhiteListMapUrl, StandardCharsets.UTF_8);
```

File: server/src/main/java/org/apache/druid/server/security/TLSUtils.java [hunk 4]
```diff
@@ -183,10 +183,12 @@ public static SSLContext createSSLContext(
       KeyStore trustStore = KeyStore.getInstance(trustStoreType == null
                                                ? KeyStore.getDefaultType()
                                                : trustStoreType);
-      trustStore.load(
-          new FileInputStream(trustStorePath),
-          trustStorePasswordProvider == null ? null : trustStorePasswordProvider.getPassword().toCharArray()
-      );
+      try (final FileInputStream trustStoreFileStream = new FileInputStream(trustStorePath)) {
```
Comment (leventov) [inline server/src/main/java/org/apache/druid/server/security/TLSUtils.java]: While at this PR, please replace usages of `FileInputStream`/`FileOutputStream` in the places that you have touched with `Files.newInputStream`/`Files.newOutputStream` when possible. See this comment: https://github.com/apache/incubator-druid/pull/6677#discussion_r237087342.

File: core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java (hunk 1 above)
Comment (asdf2014) [reply core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java]: Thanks, it is better.

File: core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java (hunk 2 above)
Comment (asdf2014) [reply core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java]: Changed.

File: extensions-contrib/ambari-metrics-emitter/src/main/java/org/apache/druid/emitter/ambari/metrics/WhiteListBasedDruidToTimelineEventConverter.java (hunk 3 above)
Comment (asdf2014) [reply extensions-contrib/ambari-metrics-emitter/src/main/java/org/apache/druid/emitter/ambari/metrics/WhiteListBasedDruidToTimelineEventConverter.java]: Thanks a lot :+1:

File: server/src/main/java/org/apache/druid/server/security/TLSUtils.java (hunk 4 above)
Comment (asdf2014) [reply server/src/main/java/org/apache/druid/server/security/TLSUtils.java]: Done.

**Comment by asdf2014** [comment]
> Great, thanks! How did you find all these leaks?

@leventov Thanks for your comments. The way to discover all resource leaks is the LGTM.

Before https://github.com/apache/incubator-druid/pull/8331



After https://github.com/apache/incubator-druid/pull/8331



Come on, let's get to A+ as soon as possible. I think it's time to add the LGTM shield to the README document. :muscle:

**Pull request merged**